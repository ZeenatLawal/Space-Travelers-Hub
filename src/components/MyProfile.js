import React from 'react';
import { useSelector } from 'react-redux';
import Container from 'react-bootstrap/Container';

const MyProfile = () => {
  const missions = useSelector((state) => state.missionsReducer.missions);

  const filterReserved = missions.filter((mission) => mission.reserved);

  return (
    <Container>
      <ul>
        {filterReserved.map((mission) => (
          <li className="border" key={mission.id}>{mission.name}</li>
        ))}
      </ul>
      {!filterReserved && (
        <h3>No Reserved Mission</h3>
      )}
    </Container>
  );
};
export default MyProfile;
