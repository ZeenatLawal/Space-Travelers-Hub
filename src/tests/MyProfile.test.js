import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

it('renders correctly', () => {
  const profile = renderer
    .create(<Provider store={store}><MyProfile /></Provider>)
    .toJSON();
  expect(profile).toMatchSnapshot();
});
