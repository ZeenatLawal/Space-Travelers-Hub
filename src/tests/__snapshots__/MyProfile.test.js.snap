// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="myProfile mt-4"
>
  <div
    className="rocketProfileWrap me-5"
  >
    <h4
      className="myRockets"
    >
      My Missions
    </h4>
    <ul
      className="myRocketLists"
    />
    <h5
      className="noReservation"
    >
      No mission yet!
    </h5>
  </div>
  <div
    className="rocketProfileWrap"
  >
    <h4
      className="myRockets"
    >
      My Rockets
    </h4>
    <h5
      className="noReservation"
    >
      No reservation yet!
    </h5>
  </div>
</div>
`;
